src/cskvp.rs

```diff
@@ -138,3 +137,129 @@ impl<'a> Drop for Cskvp<'a> {
         }
     }
 }
+
+struct Parser<'a> {
+    rest: Cow<'a, str>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum Value<'a> {
+    Double(Cow<'a, str>, Cow<'a, str>),
+    Single(Cow<'a, str>),
+}
+
+impl<'a> Parser<'a> {
+    fn new(s: Cow<'a, str>) -> Parser<'a> {
+        Parser {
+            rest: s,
+        }
+    }
+
+    fn next_quoted(&mut self) -> Cow<'a, str> {
+        assert!(self.rest.trim_start().starts_with('"'));
+
+        match &self.rest {
+            Cow::Borrowed(rest) => match quoted_string::parse::<TestSpec>(rest) {
+                // TODO: error handling
+                Err(e) => panic!("Invalid double-quoted string: {:?}", e),
+                Ok(parsed) => {
+                    self.rest = Cow::Borrowed(parsed.tail);
+                    quoted_string::to_content::<TestSpec>(parsed.quoted_string).unwrap()
```

Comment:
If I'm going to write my own Spec, I could just go ahead and write the whole parsing myself. It's not that hard, and would only need a single pass for parsing and unescaping instead of two passes in the library.

Regarding errors, the documentation of `to_content` states that it'll error if the string isn't double-quoted. But we already know that it's double-quoted, because it's the result of `parse`. I'm wondering why the library didn't make the `Parsed` struct's field private with accessor methods, and accept the Parsed struct in the `to_content` function.

In general I'm not too happy with the library, but decided to use it anyways, because that way we don't need to implement the parsing manually, reducing sources of bugs.